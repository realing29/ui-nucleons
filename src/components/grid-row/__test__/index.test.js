import React from 'react';
import { shallow } from 'enzyme';
import GridRow from '../';

describe('Компонент GridRow', () => {
  it('should match snapshot with defaults', () => {
    const gridRow = shallow(<GridRow />);
    expect(gridRow).toMatchSnapshot();
  });
  it('should renders correctly without props', () => {
    const gridRow = shallow(<GridRow />);
    expect(gridRow.find('.container').children()).toHaveLength(3);
    expect(gridRow.find('.container')).toHaveLength(1);
    expect(gridRow.find('.container').prop('className')).toEqual('container gutter-l content-undefined');
  });
  it('should match snapshot with custom props', () => {
    const gridRow = shallow(<GridRow columns={2} gutter='s' alignment='center' externalClass='test' tag='section' />);
    expect(gridRow).toMatchSnapshot();
  });
  it('should renders correctly with custom props', () => {
    const gridRow = shallow(<GridRow columns={2} gutter='s' alignment='center' externalClass='test' tag='section' />);
    expect(gridRow.find('section')).toHaveLength(1);
    expect(gridRow.find('section').children()).toHaveLength(2);
    expect(gridRow.find('section').prop('className')).toEqual('container test gutter-s content-center');
  });
});
