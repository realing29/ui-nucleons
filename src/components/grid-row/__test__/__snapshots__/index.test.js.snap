// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент GridRow should match snapshot with custom props 1`] = `
<section
  className="container test gutter-s content-center"
>
  <div
    key="0"
    style={
      Object {
        "backgroundColor": "#1f84db",
        "height": "30px",
        "width": "100%",
      }
    }
  />
  <div
    key="1"
    style={
      Object {
        "backgroundColor": "#1f84db",
        "height": "30px",
        "width": "100%",
      }
    }
  />
</section>
`;

exports[`Компонент GridRow should match snapshot with defaults 1`] = `
<div
  className="container gutter-l content-undefined"
>
  <div
    key="0"
    style={
      Object {
        "backgroundColor": "#1f84db",
        "height": "30px",
        "width": "100%",
      }
    }
  />
  <div
    key="1"
    style={
      Object {
        "backgroundColor": "#1f84db",
        "height": "30px",
        "width": "100%",
      }
    }
  />
  <div
    key="2"
    style={
      Object {
        "backgroundColor": "#1f84db",
        "height": "30px",
        "width": "100%",
      }
    }
  />
</div>
`;
